### ISSUE TITLE
SharpSvn fork that supports Linux/MacOS

### ISSUE THREAD

**Issue by A-Ovchinnikov-mx** [AmpScm/SharpSvn#1]
Hi,

First of all -- thank you for keeping SharpSvn alive! I just found the new SharpSvn 1.14 release on Nuget.org and I'm really happy it's not dead :)

A while ago I've started a rewrite of SharpSvn in pure C# to enable support for Linux and MacOS. In the readme of your project I know see a mention of those platforms.

If you're interested, please check the fork out: https://github.com/mendix/libsvnsharp

I've stopped at the place where I needed to build a Subversion pipeline for Windows, as for Linux it was pretty simple but that didn't work on MINGW64.

I'm not sure I'm ready to keep working on the fork, but if somebody would like to continue the work, I'm open for the discussion.

Cheers,
Alexander.

**Comment by rhuijben** [comment]
Nice work.

The .NetCore issue on Windows had been fixed by Microsoft, but I don't think support for C++/cli on the other platforms will arrive anytime soon.

Moving the build to GitHub workflows should have solved the how do I build the dependencies question...

NAnt was a sane choice when I started SharpSvn, but I'm trying to slowly switch to the vcpkg tool chain.

**Comment by A-Ovchinnikov-mx** [comment]
Yeah, from my understanding, C++/CLI is now considered some kind of legacy + its implementation is very tightly coupled with Windows specifics (whatever they are). So I don't expect C++/CLI to ever become cross-platform, especially given the effort they put into C# interop.

Speaking of this project, it started as my "crafting days" project and at some point I've been thinking of integrating it into our solution, but the plans have changed and I don't think SVN is a priority anymore. That being said, I might still do some work here and there, for example to make use of C# 9 capabilities (e.g. function pointers, native ints etc.), but if somebody in the community would like to take it over/fork it, please feel free to do so.
